src/components/Editor.jsx

```diff
@@ -121,23 +121,27 @@ class Editor extends React.Component {
     }
   }
 
+  _addMarkerToOpeningBracket() {
+    const {row, column} = this._editor.getCursorPosition();
+    const token = this._editor.session.getTokenAt(row, column);
+    this._editor.session.removeMarker(this._editor.session.$highlightMarker);
```

Comment:
Much nicer refactored into a method?

One thing I’m a little confused about—my read of the code here is that we are removing one marker, and then adding (at most) one marker on line 130. However, the effect of this code from playing around with it is that there are _two_ markers, for the beginning and end, which is not the case in the production code. Where’s the extra marker coming from?

Also, I noticed that if the code is in an invalid state and there is, for instance, an opening bracket with no matching closing bracket, the opening bracket is still highlighted. That doesn’t seem ideal. Is that reasonably easy to fix?